tests/data/test_binary_performance.py

```diff
@@ -227,3 +227,23 @@ def test_censor_threshold_exact(self, fake_seismo):
             censor_threshold=4,
         )
         assert result is None
+
+    def test_per_context_does_not_leak_state_across_iterations(self, fake_seismo):
+        # Ensure generate_analytics_data works correctly across multiple (score, target) pairs
+        # without interference between iterations due to in-place data mutation.
+        fake_seismo.event_aggregation_method = lambda target: "max"
+        result = generate_analytics_data(
+            score_columns=["score1", "score2"],
+            target_columns=["event1_Value", "event2_Value"],
+            metric="Sensitivity",
+            metric_values=[0.5],
+            per_context=True,
+            censor_threshold=3,
+        )
+
+        # Should return a DataFrame with 4 rows (2 scores x 2 targets)
+        assert result is not None
+        assert len(result) == 4
+
+        # Each row should have the same number of positives
+        assert result["Positives"].nunique() == 1
```

Comment:
Can we make this assertion even stronger? It took me a bit to understand why same number of positives was useful. I think its a bit easier to understand by enforcing what that unique value is. Adding as suggestion but likely need to update the comment too
```suggestion-1
        assert result["Positives"].unique().tolist() == [2]
```

Reply:
Updated to check `assert result["Positives"].tolist() == [2, 2, 2, 2]`. As you mentioned, uniqueness is coincidentally the case but is not exactly what we want.